src/Library/demos/Text Input/main.blp

```diff
@@ -0,0 +1,309 @@
+      }
+
+      Box {
+        orientation: vertical;
+
+        Label {
+          label: "Placeholder Text";
+          margin-bottom: 10;
+        }
+
+        Entry {
+          placeholder-text: "Enter text here";
+          buffer: bind entry.buffer;
+        }
+      }
+
+      Box {
+        orientation: vertical;
+
+        Label {
+          label: "Icons";
+          margin-bottom: 10;
+        }
+
+        Entry icon_entry {
+          primary-icon-name: "about-symbolic";
+          primary-icon-activatable: true;
+          primary-icon-tooltip-text: "Click on me!";
+          secondary-icon-name: "bell-symbolic";
+          secondary-icon-tooltip-text: "No click on me!";
+          buffer: bind entry.buffer;
+        }
+      }
+
+      Box {
+        orientation: vertical;
+
+        Label {
+          label: "Visiblity";
+          margin-bottom: 10;
+        }
+
+        Entry visibility_entry {
+          visibility: false;
+          secondary-icon-name: "eye-open-negative-filled-symbolic";
+          secondary-icon-activatable: true;
+          buffer: bind entry.buffer;
+        }
+      }
+    }
+
+    Box {
+      halign: center;
+      margin-top: 20;
+      margin-bottom: 20;
+
+      LinkButton{
+        label: "Tutorial";
+        uri: "https://developer.gnome.org/documentation/tutorials/beginners/components/entry.html";
+      }
+
+      LinkButton{
+        label: "API Reference";
+        uri: "https://docs.gtk.org/gtk4/class.Entry.html";
+      }
+    }
+
+    Separator {
+      margin-bottom: 20;
+    }
+
+    Label {
+      label: "Search Entry";
+      halign: start;
+
+      styles [
+        "heading",
+      ]
+    }
+
+    Label {
+      label: "An entry widget that has been tailored for searching.";
+      margin-top: 10;
+      margin-bottom: 10;
+      halign: start;
+
+      styles [
+        "dim-label",
+      ]
+    }
+
+    SearchEntry search_entry {
+      margin-bottom: 10;
+      placeholder-text: "Search for items...";
+    }
+
+    ListBox items {
+      margin-bottom: 10;
+
+      styles [
+        "boxed-list",
+      ]
+
+      ListBoxRow {
+        Label {
+          label: "Apples";
+          margin-top: 10;
+          margin-bottom: 10;
+        }
+      }
+
+      ListBoxRow {
+        Label {
+          label: "Oranges";
+          margin-top: 10;
+          margin-bottom: 10;
+        }
+      }
+
+      ListBoxRow {
+        Label {
+          label: "Bananas";
+          margin-top: 10;
+          margin-bottom: 10;
+        }
+      }
+
+      ListBoxRow {
+        Label {
+          label: "Strawberries";
+          margin-top: 10;
+          margin-bottom: 10;
+        }
+      }
+
+      ListBoxRow {
+        Label {
+          label: "Pineapples";
+          margin-top: 10;
+          margin-bottom: 10;
+        }
+      }
+
+      ListBoxRow {
+        Label {
+          label: "Blueberries";
+          margin-top: 10;
+          margin-bottom: 10;
+        }
+      }
+    }
```

Comment:
Nice adding the search entries/results :+1: 

However — it's not really how GNOME would do it because it makes content reflow. Consider using a scrolled window.

WDYT of using GtkEntryCompleton instead? There is a demo in GTK Demo
I know it's deprecated but GTK5 will take years and we can port to whatever new mechanism is added then.
I think it will make the code simpler too.

Reply:
For the content reflow, do you want me to wrap the ListBox in a ScrolledWindow?

I thought of using GtkEntryCompletion but since it was deprecated I didn't think to show it. If you're fine with it, I can replace the Search Entry with a GtkEntryCompletion demo instead.

Reply:
Yes, let's make an exception for this one and use the deprecated API since there is no replacement just yet. It was deprecated to get rid of cell renderer (the underlying backend) — not because we don't have to have this UI pattern. 

I'm sure we'll get a replacement.